Source/Core/DolphinQt2/Config/LoggerWidget.cpp

```diff
@@ -0,0 +1,496 @@
+  palette.setColor(QPalette::Text, Qt::white);
+  m_log_text->setPalette(palette);
+
+  auto cursor = m_log_text->textCursor();
+  m_log_text->setTextCursor(cursor);
+
+  // Configuration
+  m_tab_config = new QWidget();
+  auto* config_layout = new QVBoxLayout;
+  m_tab_config->setLayout(config_layout);
+
+  auto* config_verbosity = new QGroupBox(tr("Verbosity"));
+  auto* verbosity_layout = new QVBoxLayout;
+  config_verbosity->setLayout(verbosity_layout);
+  m_verbosity_notice = new QRadioButton(tr("Notice"));
+  m_verbosity_error = new QRadioButton(tr("Error"));
+  m_verbosity_warning = new QRadioButton(tr("Warning"));
+  m_verbosity_info = new QRadioButton(tr("Info"));
+
+  auto* config_outputs = new QGroupBox(tr("Logger Outputs"));
+  auto* outputs_layout = new QVBoxLayout;
+  config_outputs->setLayout(outputs_layout);
+  m_out_file = new QCheckBox(tr("Write to File"));
+  m_out_console = new QCheckBox(tr("Write to Console"));
+  m_out_window = new QCheckBox(tr("Write to Window"));
+
+  auto* config_types = new QGroupBox(tr("Log Types"));
+  auto* types_layout = new QVBoxLayout;
+  config_types->setLayout(types_layout);
+  m_types_toggle = new QPushButton(tr("Toggle All Log Types"));
+  m_types_list = new QListWidget;
+
+  for (int i = 0; i < LogTypes::NUMBER_OF_LOGS; i++)
+  {
+    QListWidgetItem* widget = new QListWidgetItem(QString::fromStdString(
+        LogManager::GetInstance()->GetFullName(static_cast<LogTypes::LOG_TYPE>(i))));
+    widget->setCheckState(Qt::Unchecked);
+    m_types_list->addItem(widget);
+  }
+
+  config_layout->addWidget(config_verbosity);
+  verbosity_layout->addWidget(m_verbosity_notice);
+  verbosity_layout->addWidget(m_verbosity_error);
+  verbosity_layout->addWidget(m_verbosity_warning);
+  verbosity_layout->addWidget(m_verbosity_info);
+
+  config_layout->addWidget(config_outputs);
+  outputs_layout->addWidget(m_out_file);
+  outputs_layout->addWidget(m_out_console);
+  outputs_layout->addWidget(m_out_window);
+
+  config_layout->addWidget(config_types);
+  types_layout->addWidget(m_types_toggle);
+  types_layout->addWidget(m_types_list);
+}
+
+void LoggerWidget::CreateMainLayout()
+{
+  setWidget(m_tab_widget);
+}
+
+void LoggerWidget::ConnectWidgets()
+{
+  // Configuration
+  connect(m_verbosity_notice, &QRadioButton::toggled, this, &LoggerWidget::SaveSettings);
+  connect(m_verbosity_error, &QRadioButton::toggled, this, &LoggerWidget::SaveSettings);
+  connect(m_verbosity_warning, &QRadioButton::toggled, this, &LoggerWidget::SaveSettings);
+  connect(m_verbosity_info, &QRadioButton::toggled, this, &LoggerWidget::SaveSettings);
+
+  connect(m_out_file, &QCheckBox::toggled, this, &LoggerWidget::SaveSettings);
+  connect(m_out_console, &QCheckBox::toggled, this, &LoggerWidget::SaveSettings);
+  connect(m_out_window, &QCheckBox::toggled, this, &LoggerWidget::SaveSettings);
+
+  connect(m_types_toggle, &QPushButton::clicked, [this] {
+    m_all_enabled = !m_all_enabled;
+
+    m_block_save = true;
```

Comment:
I don't want SaveSettings to be called for every single item. So I temporarily block it and save it afterwards.